metrics/metrics.go

```diff
@@ -111,14 +108,34 @@ func parseMetricsFile(path string) (*File, error) {
 	metrics := make(map[string]string)
 
 	for k, v := range tmpMetrics {
-		s, err := json.Marshal(v)
-		if err != nil {
-			l.Errorw("error during marshalling metric value to JSON, skipping",
-				zap.Any("value", v),
-				zap.Error(err))
-			return nil, err
+		switch reflect.TypeOf(v).Kind() { //nolint:exhaustive
+		case reflect.String:
+			if vs, ok := v.(string); ok {
+				// handle special case when "true/false" are written as string
+				if strings.ToLower(vs) == "true" {
+					metrics[k] = "1"
+					continue
+				} else if strings.ToLower(vs) == "false" {
+					metrics[k] = "0"
+					continue
+				}
+				metrics[k] = vs
+			}
+		case reflect.Bool:
+			if vb, ok := v.(bool); ok && vb {
+				metrics[k] = "1"
+			} else {
+				metrics[k] = "0"
+			}
+		default:
+			s, err := json.Marshal(v)
+			if err != nil {
+				l.Errorw("error during marshalling metric value to JSON, skipping",
+					zap.Any("key", k), zap.Any("value", v), zap.Error(err))
+				return nil, err
```

Comment:
It looks like you are not skipping bad metrics, but interrupt whole processing, no?

Reply:
right you are, it's better to skip it